### ISSUE TITLE
undefines is not an object (evaluating 'init.codecs')

### ISSUE THREAD

**Issue by sandeepg1220** [react-native-webrtc/react-native-webrtc#1292]
#### Expected Behavior

Viewer stream has to show

#### Observed Behavior

getting error as `undefines is not an object (evaluating 'init.codecs')`

#### Steps to reproduce the issue

`
import AWS from 'aws-sdk';
import {RTCView} from 'react-native-webrtc';
import {SignalingClient} from 'amazon-kinesis-video-streams-webrtc';
import {RTCPeerConnection} from 'react-native-webrtc';

const StartViewer = async () => {
      try {
        const configData = {
          channelARN:
            'arn:aws:kinesisvideo:XXXXXX',
          channelName: 'XXXXXXX',
          channelEndpoint: '',
          clientId: getRandomClientId(),
          role: 'VIEWER',
          region: 'XXXXX',
          credentials: {
            accessKeyId: 'XXXXXX',
            secretAccessKey: 'XXXXX/XXXX',
            sessionToken: null,
          },
          //systemClockOffset: 0,
        };
        const kinesisVideoClient = new AWS.KinesisVideo({
          region: configData.region,
          accessKeyId: configData.credentials.accessKeyId,
          secretAccessKey: configData.credentials.secretAccessKey,
          sessionToken: null,
          endpoint: null,
          correctClockSkew: true,
        });
        console.log('kinesisVideoClient', kinesisVideoClient);
        const describeSignalingChannelResponse = await kinesisVideoClient
          .describeSignalingChannel({
            ChannelName: configData.channelName,
          })
          .promise();
        configData.channelARN =
          describeSignalingChannelResponse.ChannelInfo.ChannelARN;
        console.log('[MASTER] Channel ARN: ', configData.channelARN);
        // Get signaling channel endpoints
        const getSignalingChannelEndpointResponse = await kinesisVideoClient
          .getSignalingChannelEndpoint({
            ChannelARN: configData.channelARN,
            SingleMasterChannelEndpointConfiguration: {
              Protocols: ['WSS', 'HTTPS'],
              Role: 'VIEWER',
            },
          })
          .promise();
        const endpointsByProtocol =
          getSignalingChannelEndpointResponse.ResourceEndpointList.reduce(
            (endpoints, endpoint) => {
              endpoints[endpoint.Protocol] = endpoint.ResourceEndpoint;
              return endpoints;
            },
            {},
          );
        console.log('[VIEWER] Endpoints: ', endpointsByProtocol);
        configData.channelEndpoint = endpointsByProtocol.WSS;
        const kinesisVideoSignalingChannelsClient =
          new AWS.KinesisVideoSignalingChannels({
            region: configData.region,
            accessKeyId: configData.credentials.accessKeyId,
            secretAccessKey: configData.credentials.secretAccessKey,
            sessionToken: null,
            endpoint: endpointsByProtocol.HTTPS,
            correctClockSkew: true,
          });
        console.log(
          'kinesisVideoSignalingChannelsClient',
          kinesisVideoSignalingChannelsClient,
        );
        // Get ICE server configuration
        const getIceServerConfigResponse =
          await kinesisVideoSignalingChannelsClient
            .getIceServerConfig({
              ChannelARN: configData.channelARN,
            })
            .promise();
        const iceServers = [];
        iceServers.push({
          urls: 'stun:stun.kinesisvideo.us-east-1.amazonaws.com:443',
        });
        getIceServerConfigResponse.IceServerList.forEach((iceServer) =>
          iceServers.push({
            urls: iceServer.Uris,
            username: iceServer.Username,
            credential: iceServer.Password,
          }),
        );
        console.log('[VIEWER] ICE servers: ', iceServers);
        const formValues = {
          sendVideo: true,
          sendAudio: false,
          openDataChannel: false,
          widescreen: true,
          fullscreen: false,
          useTrickleICE: true,
          natTraversalDisabled: false,
          forceTURN: false,
          endpoint: null,
          sessionToken: null,
        };
        // Create Signaling Client
        console.log('configData', configData);
        viewer.signalingClient = new SignalingClient(configData);

        const configuration = {
          iceServers,
          iceTransportPolicy: formValues.forceTURN ? 'relay' : 'all',
        };
        viewer.peerConnection = new RTCPeerConnection(configuration);

        // Poll for connection stats
        viewer.peerConnectionStatsInterval = setInterval(
          () => viewer.peerConnection.getStats().then(onStatsReport),
          1000,
        );

        viewer.signalingClient.on('open', async () => {
          console.log('[VIEWER] Connected to signaling service');

          // Create an SDP offer to send to the master
          console.log('[VIEWER] Creating SDP offer');
          await viewer.peerConnection.setLocalDescription(
            await viewer.peerConnection.createOffer({
              offerToReceiveAudio: true,
              offerToReceiveVideo: true,
            }),
          );

          // When trickle ICE is enabled, send the offer now and then send ICE candidates as they are generated. Otherwise wait on the ICE candidates.
          if (formValues.useTrickleICE) {
            console.log('[VIEWER] Sending SDP offer');
            viewer.signalingClient.sendSdpOffer(
              viewer.peerConnection.localDescription,
            );
          }
          console.log('[VIEWER] Generating ICE candidates');
        });

        viewer.signalingClient.on('sdpAnswer', async (answer) => {
          // Add the SDP answer to the peer connection
          console.log('[VIEWER] Received SDP answer');
          // CHANGE 0 to audio and 1 to video

          // answer.sdp = answer.sdp.replace(
          //   'a=group:BUNDLE 0 1',
          //   'a=group:BUNDLE audio video',
          // );
          // answer.sdp = answer.sdp.replace('a=mid:0', 'a=mid:audio');
          // answer.sdp = answer.sdp.replace('a=mid:1', 'a=mid:video');
          await viewer.peerConnection.setRemoteDescription(answer);
        });

        viewer.signalingClient.onicecandidate = (e) => {
          try {
            console.log('remotePC icecandidate:', e.candidate);
            if (e.candidate) {
              // CHANGE 0 to audio and 1 to video
              const candidate = e.candidate;
              // candidate.sdpMid = candidate.sdpMid.replace('0', 'audio');
              // candidate.sdpMid = candidate.sdpMid.replace('1', 'video');
              viewer.peerConnection.addIceCandidate(candidate);
            }
          } catch (err) {
            console.error(`Error adding localPC iceCandidate: ${err}`);
          }
        };

        viewer.signalingClient.on('close', () => {
          console.log('[VIEWER] Disconnected from signaling channel');
        });

        viewer.signalingClient.on('error', (error) => {
          console.error('[VIEWER] Signaling client error: ', error);
        });

        // Send any ICE candidates to the other peer
        viewer.peerConnection.addEventListener(
          'icecandidate',
          ({candidate}) => {
            if (candidate) {
              console.log('[VIEWER] Generated ICE candidate');

              // When trickle ICE is enabled, send the ICE candidates as they are generated.
              if (formValues.useTrickleICE) {
                console.log('[VIEWER] Sending ICE candidate');
                viewer.signalingClient.sendIceCandidate(candidate);
              }
            } else {
              console.log('[VIEWER] All ICE candidates have been generated');

              // When trickle ICE is disabled, send the offer now that all the ICE candidates have ben generated.
              if (!formValues.useTrickleICE) {
                console.log('[VIEWER] Sending SDP offer');
                viewer.signalingClient.sendSdpOffer(
                  viewer.peerConnection.localDescription,
                );
              }
            }
          },
        );

        // As remote tracks are received, add them to the remote view
        viewer.peerConnection.onaddstream = (event) => {
          console.log('[VIEWER] Received remote track');

          if (viewer.remoteStream) {
            return;
          }
          viewer.remoteStream = event.stream;
          setStream(event.stream);
          console.log('This is the event', event);
          console.log('This is the stream', event.stream);
        };
        console.log('[VIEWER] Starting viewer connection');
        viewer.signalingClient.open();
      } catch (e) {
        console.log('error::', e);
      }
    };`

#### Platform Information

* **React Native Version**: 0.67.1
* **Plugin Version**: 106.0.3
* **OS**: iOS - iPhone 14 Pro
* **OS Version**:  - iOS 16.1

**Comment by sandeepg1220** [comment]
I'm facing the above error. could you please help anyone

**Comment by sandeepg1220** [comment]
![Simulator Screen Shot - iPhone 14 Pro - 2023-01-16 at 23 47 40](https://user-images.githubusercontent.com/57384948/212747661-bddaa1f8-6bd4-49f4-a959-3a6d12ad787e.png)

**Comment by sandeepg1220** [comment]
@saghul Could you please help me on this issue as you closed this issue

**Comment by sandeepg1220** [comment]
I'm able to get the audio but video stream is not coming. Is there anything missed

**Comment by sandeepg1220** [comment]
In the above example I have commented the sdp answer and remote ice candidate audio, video orders. If we remove the commented code getting unhandled exception has 

`Possible Unhandled Promise Rejection (id: 0):
Object {
  "message": "Failed to set remote answer sdp: The order of m-lines in answer doesn't match order in offer. Rejecting answer.",
  "name": "SetRemoteDescriptionFailed",
}
`

**Comment by saghul** [comment]
Can you please update to 106.0.4? We made some changes around that code and I'd like to see if it still fails.

**Comment by saghul** [comment]
> In the above example I have commented the sdp answer and remote ice candidate audio, video orders. If we remove the commented code getting unhandled exception has
> 
> `Possible Unhandled Promise Rejection (id: 0): Object { "message": "Failed to set remote answer sdp: The order of m-lines in answer doesn't match order in offer. Rejecting answer.", "name": "SetRemoteDescriptionFailed", } `

The error message is pretty self-explanatory here.

**Comment by sandeepg1220** [comment]
> Can you please update to 106.0.4? We made some changes around that code and I'd like to see if it still fails.

Sure I will check and let you know

**Comment by sandeepg1220** [comment]
@saghul  Same error is getting 

` LOG  rn-webrtc:pc:DEBUG 0 ontrack +7ms
 ERROR  TypeError: undefined is not an object (evaluating 'init.codecs')`

**Comment by sandeepg1220** [comment]
Audio is getting but video stream is not coming. Is there anything we were missing related to video ?

**Comment by sandeepg1220** [comment]
@saghul we have checked with version `1.106.1` the error is not getting but still video is not streaming.

**Comment by sandeepg1220** [comment]
@saghul Any update

**Comment by sandeepg1220** [comment]
@saghul Any suggestions for which version of react native need to use to work this. I'm completely blocked from 20days. Please help me.

**Comment by TheRealNate** [comment]
+1 on this issue. Basically tried the exact same code that works on web and it's failing. I'm trying to hook into a one-way WebRTC stream (a security camera) where the app receives video and audio but does not transmit.

**Comment by saghul** [comment]
Can you please try this PR? https://github.com/react-native-webrtc/react-native-webrtc/pull/1302 I think I caught it.

**Comment by sandeepg1220** [comment]
> undefines is not an object (evaluating 'init.codecs')

The above issue is resolved in [#1302 ](https://github.com/react-native-webrtc/react-native-webrtc/pull/1302)

But the video stream is still not happening only audio is coming.

**Comment by sandeepg1220** [comment]
I will open a new issue for video stream. Thank you @saghul
